diff --git a/boost/config/compiler/visualc.hpp b/boost/config/compiler/visualc.hpp
index aa4584d..4757ee0 100644
--- a/boost/config/compiler/visualc.hpp
+++ b/boost/config/compiler/visualc.hpp
@@ -310,8 +310,8 @@
 #endif
 
 //
-// last known and checked version is 1700 (VC11, aka 2011):
-#if (_MSC_VER > 1700)
+// last known and checked version is 1800 (VC13, aka 2013):
+#if (_MSC_VER > 1800)
 #  if defined(BOOST_ASSERT_CONFIG)
 #     error "Unknown compiler version - please run the configure tests and report the results"
 #  else
